src/test/java/io/xstefank/wildfly/bot/utils/JsonHandler.java

```diff
@@ -0,0 +1,141 @@
+package io.xstefank.wildfly.bot.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Class responsible for handling and changing properties of the JSON template file.
+ */
+public class JsonHandler {
+
+    private static String PULL_REQUEST = "pull_request";
+    private static String COMMITS = "commits";
+    private static String BODY = "body";
+    private static String TITLE = "title";
+    private static String SHA = "sha";
+    private static String HEAD = "head";
+    private static String ACTION = "action";
+
+    private static JsonNode file;
+
+    private JsonHandler(JsonHandlerBuilder jsonHandlerBuilder) {
+        this.file = jsonHandlerBuilder.jsonFile;
+    }
+
+    /**
+     * Returns the SHA of the first commit in the GitHub PR payload.
+     *
+     * @return the SHA of the first commit in the GitHub PR payload.
+     */
+    public String getCommitSHA() {
+        return file.get(PULL_REQUEST).get(HEAD).get(SHA).textValue();
+    }
+
+    /**
+     * Returns the JSON file using the {@link JsonNode#toPrettyString()}.
+     * The {@code toString} method returns the file as a single line String. The {@code toPrettyString} returns the
+     * file in easily readable way.
+     *
+     * @return the Spring representation of the JSON file.
+     */
+    public String getJsonString() {
+        return file.toPrettyString();
+    }
+
+    /**
+     * Builder for the JsonHandler class.
+     */
+    public static class JsonHandlerBuilder {
+
+        private final JsonNode jsonFile;
+
+        /**
+         * Builder for the JsonHandler class.
+         *
+         * @param file File instance of the JSON file.
+         * @throws IOException if an error occurred during the JSON file obtaining.
+         */
+        public JsonHandlerBuilder(File file) throws IOException {
```

Comment:
We might consider making the builder constructor as static method to avoid using `new` and perhaps call the method as `create`.

Reply:
+1 `newBuilder()` or just `builder()`